src/librustdoc/clean/utils.rs

```diff
@@ -431,39 +431,49 @@ crate fn get_auto_trait_and_blanket_impls(
     auto_impls.into_iter().chain(blanket_impls)
 }
 
-crate fn register_res(cx: &mut DocContext<'_>, res: Res) -> DefId {
-    debug!("register_res({:?})", res);
-
-    let (did, kind) = match res {
-        Res::Def(DefKind::Fn, i) => (i, ItemType::Function),
-        Res::Def(DefKind::TyAlias, i) => (i, ItemType::Typedef),
-        Res::Def(DefKind::Enum, i) => (i, ItemType::Enum),
-        Res::Def(DefKind::Trait, i) => (i, ItemType::Trait),
+/// The goal of this function is to extract the [`DefId`] and the [`ItemType`] from the given `res`.
+///
+/// In case this not a [`Res::Def`] or a [`Res::SelfTy`], the returned [`ItemType`] will be `None`.
```

Comment:
This is not an accurate description, register_res ignores some `Res::Defs`. This returns the DefId of the page the `res` will be documented on, and registers the DefId if the item will be documented. If the item won't get its own page, it returns the original id. See https://github.com/rust-lang/rust/pull/84880.

Reply:
For `register_res`, it's the case, but not for `register_res_inner`. I think the function name is simply bad. Do you have suggestions?

Reply:
Actually I removed this function because I wasn't using it anymore. Problem solved. :)